File: core/src/nl/t64/game/rpg/subjects/PartySubject.java
package nl.t64.game.rpg.subjects;

import java.util.ArrayList;
import java.util.List;


public class PartySubject {

    private final List<PartyObserver> observers = new ArrayList<>();

    public void addObserver(PartyObserver observer) {
        observers.add(observer);
    }

    public void removeObserver(PartyObserver observer) {
        observers.remove(observer);
    }

    public void removeAllObservers() {
        observers.clear();
    }

    public void notifyHeroDismissed() {
        observers.forEach(PartyObserver::onNotifyHeroDismissed);
    }

}

File: core/src/nl/t64/game/rpg/subjects/PartySubject.kt
package nl.t64.game.rpg.subjects


class PartySubject {

    private val observers: MutableList<PartyObserver> = ArrayList()

    fun addObserver(observer: PartyObserver) {
        observers.add(observer)
    }

    fun removeObserver(observer: PartyObserver) {
        observers.remove(observer)
    }

    fun removeAllObservers() {
        observers.clear()
    }

    fun notifyHeroDismissed() {
        observers.forEach { it.onNotifyHeroDismissed() }
    }

}
